### PR TITLE
#2691 Extracting Kafka-based integration tests into their own module

### PR DESCRIPTION
**Pull request by gunnarmorling**
I've copied the _integration-tests/main/pom.xml_ and stripped it down as much as possible for the new module. I wasn't quite sure about the Graal-related parts, so please let me know if there's anything missing. Thanks!

### CODE REVIEW

File: integration-tests/kafka/pom.xml [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+                        <groupId>${project.groupId}</groupId>
+                        <artifactId>quarkus-maven-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>native-image</id>
+                                <goals>
+                                    <goal>native-image</goal>
+                                </goals>
+                                <configuration>
+                                    <addAllCharsets>true</addAllCharsets>
```
Comment (cescoffier) [inline integration-tests/kafka/pom.xml]: any reason for this?

File: integration-tests/kafka/pom.xml [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+            <plugin>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>quarkus-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <goals>
+                            <goal>build</goal>
+                        </goals>
+                        <configuration>
+                            <uberJar>true</uberJar>
```
Comment (cescoffier) [inline integration-tests/kafka/pom.xml]: any reason why we need the uberJar ?

File: integration-tests/kafka/pom.xml [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            <groupId>org.apache.kafka</groupId>
+            <artifactId>kafka_2.12</artifactId>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <resources>
+            <resource>
+                <directory>src/main/resources</directory>
```
Comment (cescoffier) [inline integration-tests/kafka/pom.xml]: Is filtering explicitly used?

File: integration-tests/kafka/pom.xml [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        <artifactId>quarkus-integration-tests-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-integration-test-kafka</artifactId>
+    <name>Quarkus - Integration Tests - Kafka</name>
+    <description>The Kafka integration tests module</description>
```
Comment (cescoffier) [inline integration-tests/kafka/pom.xml]: Apache Kafka ;-)

**Review by cescoffier — DISMISSED** [review]
a few very minor comments.

File: integration-tests/kafka/pom.xml (hunk 3 above)
Comment (gunnarmorling) [reply integration-tests/kafka/pom.xml]: Good question, probably not. Took all that when copying the POM from the _main_ module. Will rework and push.

File: integration-tests/kafka/pom.xml (hunk 1 above)
Comment (gunnarmorling) [reply integration-tests/kafka/pom.xml]: Removed.

File: integration-tests/kafka/pom.xml (hunk 2 above)
Comment (gunnarmorling) [reply integration-tests/kafka/pom.xml]: Removed.

File: integration-tests/kafka/pom.xml (hunk 3 above)
Comment (gunnarmorling) [reply integration-tests/kafka/pom.xml]: Removed.

**Comment by gunnarmorling** [comment]
@cescoffier, pushed one more commit to address your remarks.

**Comment by gsmet** [comment]
@gunnarmorling could you squash the third commit with the first one so that we avoid unnecessary commits?

Apart from that, it looks good to me!

**Comment by gunnarmorling** [comment]
> could you squash the third commit with the first one so that we avoid unnecessary commits?

Done, @gsmet. I had left it separately initially for the sake of easier reviewing.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Yes, that's what I suspected.

Merged, thanks! This one was bugging me for a while.

**Pull request merged**